web3.js/src/connection.ts

```diff
@@ -2775,38 +2778,54 @@ export class Connection {
     return res.result;
   }
 
+  confirmTransaction(
+    signature: TransactionSignatureBlockhash,
+    commitment?: Commitment,
+  ): Promise<RpcResponseAndContext<SignatureResult>>;
+
   /**
-   * Confirm the transaction identified by the specified signature.
+   * @deprecated Using confirmTransaction with only a signature is deprecated.
    */
-  async confirmTransaction(
+  // eslint-disable-next-line no-dupe-class-members
+  confirmTransaction(
     signature: TransactionSignature,
     commitment?: Commitment,
+  ): Promise<RpcResponseAndContext<SignatureResult>>;
+
+  // eslint-disable-next-line no-dupe-class-members
+  async confirmTransaction(
+    signature: TransactionSignatureBlockhash | TransactionSignature,
```

Comment:
See my comment in this review about why this param should probably be named `config` or `strategy` now.